extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisSequenceManager.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.polaris.core.persistence.models.ModelSequenceId;
+import org.eclipse.persistence.internal.jpa.EntityManagerImpl;
+import org.eclipse.persistence.platform.database.DatabasePlatform;
+import org.eclipse.persistence.platform.database.PostgreSQLPlatform;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Used to generate sequence IDs for Polaris entities. If the legacy `POLARIS_SEQ` generator is
+ * available it will be used then cleaned up. In all other cases the `POLARIS_SEQUENCE` table is
+ * used directly.
+ */
+public class PolarisSequenceManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PolarisSequenceManager.class);
+
+  private static AtomicBoolean sequenceCleaned = new AtomicBoolean(false);
+
+  /* Get the database platform associated with the `EntityManager` */
+  private static DatabasePlatform getDatabasePlatform(EntityManager session) {
+    EntityManagerImpl entityManagerImpl = session.unwrap(EntityManagerImpl.class);
+    return entityManagerImpl.getDatabaseSession().getPlatform();
+  }
+
+  private static void removeSequence(EntityManager session) {
+    LOGGER.info("Renaming legacy sequence `POLARIS_SEQ` to `POLARIS_SEQ_UNUSED`");
+    String renameSequenceQuery = "ALTER SEQUENCE POLARIS_SEQ RENAME TO POLARIS_SEQ_UNUSED";
+    session.createNativeQuery(renameSequenceQuery).executeUpdate();
+  }
+
+  private static synchronized Optional<Long> getSequenceId(EntityManager session) {
+    DatabasePlatform databasePlatform = getDatabasePlatform(session);
+    if (databasePlatform instanceof PostgreSQLPlatform) {
+      Optional<Long> result = Optional.empty();
+      if (!sequenceCleaned.get()) {
+        try {
+          LOGGER.info("Checking if the sequence POLARIS_SEQ exists");
+          String checkSequenceQuery =
+              "SELECT COUNT(*) FROM information_schema.sequences WHERE sequence_name IN ('polaris_seq', 'POLARIS_SEQ')";
+          int sequenceExists =
+              ((Number) session.createNativeQuery(checkSequenceQuery).getSingleResult()).intValue();
+
+          if (sequenceExists > 0) {
+            LOGGER.info("POLARIS_SEQ exists, calling NEXTVAL");
+            long queryResult =
+                (long) session.createNativeQuery("SELECT NEXTVAL('POLARIS_SEQ')").getSingleResult();
+            result = Optional.of(queryResult);
+          } else {
+            LOGGER.info("POLARIS_SEQ does not exist, skipping NEXTVAL");
+          }
+        } catch (Exception e) {
+          LOGGER.info(
+              "Encountered an exception when checking sequence or calling `NEXTVAL('POLARIS_SEQ')`",
+              e);
+        }
+        if (result.isPresent()) {
+          removeSequence(session);
+        }
+        sequenceCleaned.set(true);
+      }
+      return result;
+    } else {
+      LOGGER.info("Skipping POLARIS_SEQ / NEXTVAL check for platform " + databasePlatform);
+      return Optional.empty();
+    }
+  }
```

Comment:
It's not strictly Postgres that supports SEQUENCE -- H2 also supports it, and other DBs support similar logic with different syntax.

Really the goal here is to patch this ASAP and then migrate all new usage to the table. Eventually, we can yank all this postgres-specific code.

Reply:
Sorry I wasn't clear in my first comment. The code is mainly for migration in case anyone is already using Postgres in production. We have two approaches:

1. Ignore it, assuming no one is actually using it in production yet—especially since Apache Polaris hasn't had its first release.
2. Wrap the migration logic into a method so we can easily remove it in the future.

I prefer the second option. With that, we could make the following changes:

1. Call the migration code only in the initialization method, instead of reusing it in `getNewId()`, since the migration happens only once. In case of migration failure, we should just throw.
2. Simplify `getNewId()`, as we don't need to distinguish if it's migrated or not. Specifically, we can remove the line `getSequenceId(session).ifPresent(modelSequenceId::setId);`.
3. Eliminate `PolarisSequenceUtil::sequenceCleaned` and `PolarisEclipseLinkStore::initialized`, as we throwed in case of migration failure, there is no need to check if migration actually happened.

Reply:
The migration code is `getSequenceId` itself. Eventually, it can be replace with `return Optional.empty()`.

Reply:
I've moved most of this into `initialize` and shifted some things around -- let me know if this more closely matches your expectations